Android.mk

```diff
@@ -145,6 +145,7 @@ LOCAL_EXPORT_C_INCLUDE_DIRS     := \
 
 LOCAL_CPPFLAGS                                                              := \
     -std=c++11                                                                 \
+    -Wno-error=non-virtual-dtor                                                \
```

Comment:
We'd better not suppress the warnings. Is there a way to get rid of the warning? Not very sure why such warning would occur in this PR, can we avoid it by declaring a virtual destructor?

Reply:
This is only required when `delete` is used on a class. And there's another flag `-Werror=delete-non-virtual-dtor` enables warning on this, which is already included in `-Wall`.